src/libs/ReportUtils.ts

```diff
@@ -3459,14 +3473,78 @@ function shouldHideReport(report: OnyxEntry<Report>, currentReportId: string): b
     return parentReport?.reportID !== report?.reportID && !isChildReportHasComment;
 }
 
+/**
+ *  Checks if there are any violations belonging to the transaction in the transactionsViolations Onyx object
+ *  then checks that the violation is of the proper type
+ */
+function transactionHasViolation(transactionID: string, transactionViolations?: TransactionViolations): boolean {
+    const violations: TransactionViolation[] = transactionViolations ? transactionViolations[transactionID] || [] : [];
+    if (!violations) {
+        return false;
+    }
+    return violations.some((violation: TransactionViolation) => violation.type === 'violation');
+}
+
+/**
+ *  Checks to see if a report's parentAction is a money request that contains a violation
+ *  This only applies to report submitter and for reports in the open and processing states
+ */
+
+function transactionThreadHasViolations(
+    report: Report,
+    canUseViolations: boolean,
+    transactionViolations?: TransactionViolations,
+    reportActions?: OnyxCollection<ReportActions> | null,
+    parentReportAction?: ReportAction | null,
```

Comment:
Why do we need any of these params? Just curious since I probably missed something, but the original design only had `report` as a param. I could see us adding `canUseViolations` since returning early based on that value can be convenient, but we could also check that before calling this function

Reply:
After moving away from `Onyx.Connect` within the `ReportUtils` file in favor of `withOnyx`, we need these values to be passed down from the components where they're called. I could remove the parentReportAction param, I had just included that because there are some instances where that was already calculated.

Reply:
Let's remove it. Can we make `reportActions` mandatory?